### ISSUE TITLE
readPackage hook can not effect scripts.postinstall

### ISSUE THREAD

**Issue by wclr** [pnpm/pnpm#3071]
### pnpm version: 5.15.0

I need to disable `postinstall` script of one of the packages (but I can not use `--ignore-script` as it effects all), so wanted to do this via `readPackage` hook, but it doesn't seem to have an effect, postinstall is still running.

### Expected behavior:

I would expect that `postinstal`l behaviour could be changed for particular package script via readPackage hook.

**Comment by zkochan** [comment]
Let's just support this: https://yarnpkg.com/configuration/manifest#dependenciesMeta.built

**Comment by wclr** [comment]
So, I believe, it is not supported yet? No way to disable post install for a package currently?

**Comment by zkochan** [comment]
There's no way. Only to ignore all with `ignore-scripts`

**Comment by wclr** [comment]
Ok, managed to disable it by modifying `bin` prop in the readPackage hook (though this works just for my particular case). 

I just wonder why `scripts.postinstall` is not affected by readPackage hook? Why it is read before the hook?

**Comment by zkochan** [comment]
We can add support for a new property for disabling specific package's scripts. For instance:

```json
{
  "pnpm": {
    "neverBuildDependencies": ["fsevents"]
  }
}
```

**Comment by wclr** [comment]
This would be nice.

**Comment by wclr** [comment]
This is not released yet?

**Comment by zkochan** [comment]
not released yet

**Comment by wclr** [comment]
I just realized that in my case problem is that I need to run `postinstall` in one environment (on the host) and don't need docker container where the same packages are installed. So I implement this logic in `readHook` (which I failed to do and created this issue). 

With this change, I would need to modify (remove) not side effect package's `postinstall`, but root's package `pnpm.neverBuiltDependencies` entry. Is it going to work? Wich it be handled after `readPackage` hook?

**Comment by zkochan** [comment]
this is a way too complex scenario. It will not work.

**Comment by wclr** [comment]
I don't get why not handle `readPakage` hook before any handing pkg file operations? Well, some fields like `name` could be considered immutable not to break everything, but why `postinstall` is not or other entries can not be modified by hook effectually?

**Comment by zkochan** [comment]
readPackage is performed on the metadata package.json that comes from the registry via a json response. Scripts are read from the package.json file that is in the package's archive. npm claims that those two files don't always match.

Even if we will run readPackage twice for every package, scripts is not the only way builds are described. Builds are also performed if the package has a `hooks` directory with scripts.

I think you overcomplicate things. You can just ignore the build for your package and trigger it manually in the environment where you need it.

**Comment by wclr** [comment]
> You can just ignore the build for your package and trigger it manually in the environment where you need it.

Yes, this is another scenario for solving the problem, but it still requires scripting to make things automatic. Introducing the need for manual procedures is not a good way to go.

**Comment by wclr** [comment]
> readPackage is performed on the metadata package.json 

Then it probably will be useful for users to reflect in docs which files can be effectually modified by the hook.

**Comment by cyrus-za** [comment]
Any way we can get this feature done sooner rather than later?
https://github.com/pnpm/pnpm/issues/3071#issuecomment-761851972

**Comment by zkochan** [comment]
That features was implemented in v5.16

https://pnpm.io/package_json#pnpmneverbuiltdependencies
